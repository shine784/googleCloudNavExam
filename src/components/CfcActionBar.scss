.cfc-action-bar {
    border-bottom: 1px solid grey;
}