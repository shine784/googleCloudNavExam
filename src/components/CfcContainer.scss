.cfc-container{
    box-sizing: border-box;
    display: flex;
    flex-direction: row;
    background-color: #ededed;
    height: 100%;
}
@media screen and (max-width: 768px) {
    .cfc-container{
        display: flex;
    }
  }
  
@media screen and (max-width: 768px) {
    .cfc-container{
        display: block;
    }
}