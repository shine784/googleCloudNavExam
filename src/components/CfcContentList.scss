.cfc-content-list{
    li {
        min-height:30px;
        height: auto;
        border: 1px solid rgb(39, 15, 255);
        background-color: blanchedalmond;
    }
}