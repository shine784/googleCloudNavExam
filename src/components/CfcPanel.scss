.cfc-panel{
    flex:1;
    display: flex;
    flex-direction: column;
    overflow: visible;
    position: relative;
    min-height:0;
    border: 1px solid rgb(214, 214, 214);
    height:100%;
    &.left{
        flex:0 1 auto;
        width:255px;
    }
    &.right{
       
    }
}


@media screen and (max-width: 768px) {
    .cfc-panel.left{
      //display: block;
      width: 100%;
      height: auto;
    }
  }
  
@media screen and (max-width: 768px) {
    .cfc-panel.left{
      //display: none;
    }
}