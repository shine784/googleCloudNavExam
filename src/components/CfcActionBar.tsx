import { observer } from "mobx-react-lite";
import React from 'react';
import "./CfcActionBar.scss";
interface Props {
    title: String
}

const CfcActionBar = observer((props: Props) => {
    return (
        <div className="cfc-action-bar">
            <h1>{props.title}</h1>
        </div>
    )
})

export default CfcActionBar;