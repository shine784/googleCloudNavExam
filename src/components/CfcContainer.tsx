import { observer } from "mobx-react-lite";
import React from 'react';
import "./CfcContainer.scss";
const CfcContainer = observer((prop) => {
    return (
        <div data-testid="cfc-container" className="cfc-container">
            {prop.children}
        </div>
    )
})

export default CfcContainer;