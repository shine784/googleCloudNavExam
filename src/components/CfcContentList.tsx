import { observer } from "mobx-react-lite";
import React, { useEffect } from 'react';
import "./CfcContentList.scss";
interface ABC{
    title:String
}

interface Props {
    list:any[]
}
const CfcContentList = observer(({list}:Props) => {
    //const store = useStore();
    //const kubernetesModel = store.getKubernetesModel(); 
    return (
        <div data-testid="cfc-content-list" className="cfc-content-list">
            <ul>
            {
            list.map((d:any,i:number)=>{
                return <li key={i}>{d.title}</li>
            })
            }
            </ul>
        </div>
    )
})

export default CfcContentList;