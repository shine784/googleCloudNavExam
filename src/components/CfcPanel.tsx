import { observer } from "mobx-react-lite";
import React from 'react';
import classNames from 'classnames/bind';
import "./CfcPanel.scss";

interface Props{
    pos:String,
    children: React.ReactNode
}

const CfcPanel = observer((props:Props) => {
    const panelClass = classNames({
        "cfc-panel": true,
        [`${props.pos}`]: true,
      });
    return (
        <div className={panelClass}>
            {props.children}
        </div>
    )
})

export default CfcPanel;