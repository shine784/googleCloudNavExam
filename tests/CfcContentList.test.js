import React from "react";
import CfcContentList from "../src/components/CfcContentList";
import CfcActionBar from "../src/components/CfcActionBar";
import CfcSectionTitle from "../src/components/CfcSectionTitle";
import CfcContainer from "../src/components/CfcContainer";
import CfcPanel from "../src/components/CfcPanel";
import "@testing-library/jest-dom";
//import CfcNavigationArea from '../src/components/CfcNavigationArea'
import { render, fireEvent, wrapper } from "@testing-library/react";

beforeAll(() => {});

test("CfcContentList component", () => {
	const { getByTestId, getByText, baseElement, container } = render(
		<CfcContentList list={[{ title: "123" }, { title: "456" }]} />
	);
	expect(getByText("123")).toBeInTheDocument();
	expect(getByText("456")).toBeInTheDocument();

	expect(getByTestId("cfc-content-list")).toMatchSnapshot();
});

test("CfcActionBar component", () => {
	const { getByText, baseElement, container } = render(<CfcActionBar title="muyaho" />);
	expect(getByText("muyaho")).toBeInTheDocument();
	expect(getByText("muyaho")).toMatchSnapshot();
});

test("CfcSectionTitle component", () => {
	const { getByText, baseElement, container } = render(<CfcSectionTitle title="muyaho" />);
	expect(getByText("muyaho")).toBeInTheDocument();
	expect(getByText("muyaho")).toMatchSnapshot();
});

test("CfcContainer component", () => {
	const { getByTestId,getByText, baseElement, container } = render(
		<CfcContainer>
			<CfcContentList list={[{ title: "123" }, { title: "456" }]} />
		</CfcContainer>
	);
    expect(getByTestId("cfc-container")).toMatchSnapshot();
});

test("CfcPanel component", () => {
	const { getByText, baseElement, container } = render(<CfcPanel />);
});
