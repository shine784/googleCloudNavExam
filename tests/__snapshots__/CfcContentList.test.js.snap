// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CfcActionBar component 1`] = `
<h3>
  muyaho
</h3>
`;

exports[`CfcContainer component 1`] = `
<div
  class="cfc-container"
  data-testid="cfc-container"
>
  <div
    class="cfc-content-list"
    data-testid="cfc-content-list"
  >
    <ul>
      <li>
        123
      </li>
      <li>
        456
      </li>
    </ul>
  </div>
</div>
`;

exports[`CfcContentList component 1`] = `
<div
  class="cfc-content-list"
  data-testid="cfc-content-list"
>
  <ul>
    <li>
      123
    </li>
    <li>
      456
    </li>
  </ul>
</div>
`;

exports[`CfcSectionTitle component 1`] = `
<h3>
  muyaho
</h3>
`;
